lib/cloud/azure/vm.go

```diff
@@ -327,6 +331,36 @@ func (c *runCommandClient) Run(ctx context.Context, req RunCommandRequest) error
 		return trace.Wrap(err)
 	}
 
-	_, err = poller.PollUntilDone(ctx, nil /* options */)
-	return trace.Wrap(err)
+	_, err = poller.PollUntilDone(ctx, &runtime.PollUntilDoneOptions{Frequency: 10 * time.Second})
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	// note: we are not guaranteed to receive the output of the command above if the req.Name is not unique.
```

Comment:
Did you mean req.VMName? I'm also assuming that this would depend on `runCommandName`, and it's only still true because the command name is a constant, correct? If so, this comment may benefit from including that, too; otherwise it would be a bit confusing.

Reply:
No, `req.Name` is right. Two discovery agents may run the same installation script at the same time, so there is a race condition involved. I want to solve that race condition in the future, but so far we depend on the installation being "eventually consistent" (or so we hope).